lib/parsers/iso_base_media_file_format/box.rb

```diff
@@ -0,0 +1,80 @@
+module FormatParser
+  module ISOBaseMediaFileFormat
+    class Box < Struct.new(:type, :position, :size, :fields, :children)
+      def initialize(type, position, size, fields = nil, children = nil)
+        super
+        self.fields ||= {}
```

Comment:
You don't need an `initialize` for this; you can simply define a:

```
def fields
  @fields ||= {}
end
```

Or, seeing that you need so much more customisation on top of what Struct gives, you might be better off not using it at all (you're only saving a few assignments when initialising the class).

Reply:
Hmm, removing the `initialize` def and adding getter methods for `fields` and `children` as you suggested causes several test failures - so it's changing the behaviour and I'm not really sure how.

We could ditch Struct, but it gives us more than just a constructor. For example, we get a more useful string representation a `to_h` method and equality operators, all of which are useful for debugging. Also, it gives us a `dig` method which works well with your suggestion down below to avoid chaining `&.` operators - so I'd rather keep it as-is.

Reply:
Probably you need `self.fields ||= {}` instead of `@fields ||= {}`. Also, if you're not using any args in `initialize`, you can just `initialize(*args)` them.

I see, if there are multiple benefits of the Struct, let's keep it.